### PR TITLE
Lvnod 60 create mailer

### PR DESCRIPTION
**Pull request by shyyka**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-XXX)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-XXX)

## Code reviewers

- [x] @yurii-kokhalevych
- [x] @denisbodnar
- [x] @yuliannakorol

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of issue

ToDo

## Summary of change

ToDo
## CHECK LIST
- [x]  СI passed
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by yurii-kokhalevych — DISMISSED** [review]

File: server/templates/mailer/activation.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html>
+<head>
+	<title>Da-Mi</title>
```
Comment (denisbodnar) [inline server/templates/mailer/activation.html]: Use spaces, not tabs

**Review by denisbodnar — DISMISSED** [review]

File: server/helper/mailer.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const nodemailer = require('nodemailer');
```
Comment (4asof4ik) [inline server/helper/mailer.js:3]: add nodemailer into package.json

File: server/helper/mailer.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+const handlebars = require('handlebars');
+const fs = require('fs');
+const path = require('path');
+
+const secret = require(path.resolve('./config', 'secretkey.json'));
+let readHTMLTemplate = function(_name, callback) {
+    fs.readFile(path.join('./server/templates/mailer/' + _name + '.html'), {
+            encoding: 'utf-8'
+          },
+          function(err, html) {
```
Comment (4asof4ik) [inline server/helper/mailer.js]: use ES6 => func

File: server/helper/mailer.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+const nodemailer = require('nodemailer');
+const smtpTransport = require('nodemailer-smtp-transport');
+const handlebars = require('handlebars');
+const fs = require('fs');
+const path = require('path');
+
+const secret = require(path.resolve('./config', 'secretkey.json'));
+let readHTMLTemplate = function(_name, callback) {
+    fs.readFile(path.join('./server/templates/mailer/' + _name + '.html'), {
```
Comment (4asof4ik) [inline server/helper/mailer.js]: should be views but not templates

File: server/helper/mailer.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                to: _data.email,
+                subject: _data.subject,
+                html: htmlToSend,
+                attachments: [{
+                    filename: _data.img,
+                    path: ('./server/templates/mailer/img/' + _data.img),
+                    cid: 'bonsai@kreata.ee'
+                  }]
+              };
+            transport.sendMail(mailOptions, function(error, response) {
```
Comment (4asof4ik) [inline server/helper/mailer.js]: ES6

File: server/helper/mailer.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    secure: false,
+    auth: {
+        user: secret.gmail.user,
+        pass: secret.gmail.pass
+      }
+  }));
+
+module.exports = {
+    _send(_data, _template) {
+      readHTMLTemplate(_template, function(err, html) {
```
Comment (4asof4ik) [inline server/helper/mailer.js]: ES6

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

File: server/helper/mailer.js (hunk 2 above)
Comment (shyyka) [reply server/helper/mailer.js:3]: done.

**Review by denisbodnar — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

File: server/helper/mailer.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      };
+
+    template.render(locals, (err, sendMail) => {
+
+      if (err) {
+        return console.error(err);
+      }
+
+      let mailOptions = {
+        from: '"Da-Mi" <no-reply@da-mi.com>',
```
Comment (omaik) [inline server/helper/mailer.js]: How do you plan to make it work on production?
Have you bought SES?

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

File: server/helper/mailer.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+const smtpTransport = require('nodemailer-smtp-transport');
+const handlebars = require('handlebars');
+const EmailTemplate = require('email-templates').EmailTemplate;
+const path = require('path');
+
+const secret = require(path.resolve('./config', 'secretkey.json'));
+
+let transport = nodemailer.createTransport(smtpTransport({
+    host: 'smtp.gmail.com',
+    port: 587,
```
Comment (lhalam) [inline server/helper/mailer.js]: host and port move to config

File: server/helper/mailer.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        url: url,
+        eventName: _data.eventName,
+        date: _data.date,
+        eventDescription: _data.eventDescription
+      };
+
+    template.render(locals, (err, sendMail) => {
+
+      if (err) {
+        return console.error(err);
```
Comment (lhalam) [inline server/helper/mailer.js]: use logger

File: server/helper/mailer.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+      }
+
+      let mailOptions = {
+        from: '"Da-Mi" <dami.serv@gmail.com>',
```
Comment (lhalam) [inline server/helper/mailer.js]: move to config

File: server/helper/mailer.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+        to: _data.email,
+        subject: _data.subject,
+        html: sendMail.html,
+        text: sendMail.text,
+        attachments: [{
+          filename: _data.img,
+          path: ('./server/views/emails/img/' + _data.img),
+          cid: 'bonsai@kreata.ee'
```
Comment (lhalam) [inline server/helper/mailer.js]: move to config or constant

**Review by lhalam — COMMENTED** [review]
todo

**Review by denisbodnar — DISMISSED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Review by denisbodnar — APPROVED** [review]

**Pull request merged**